### DISCUSSION TITLE
Data for Singapore

### DISCUSSION THREAD

**Discussion by hxchua** [neherlab/covid19_scenarios#700] [General] [answered]
Hi! I'm looking to contribute data for Singapore as I've been collating and updating the data daily. I noticed the case count data required is [time, cases, deaths, hospitalized, ICU, recovered]. However, in the context of Singapore, we have people who are not hospitalized but are isolated and cared for at community facilities - as there's not enough beds in the hospitals to go around. These are those who have mild symptoms, or are clinically well but still test positive for COVID-19. They contribute to the count of total cases. How should I be including these people in the case count data?

The raw data can be found [here](https://docs.google.com/spreadsheets/d/1gFTNs_GtnTIyyVWXmsQxwdZpGbyicZM2HJcXvCf4b3k/edit?usp=sharing).  
More details about the dataset can be found here: https://data.world/hxchua/covid-19-singapore 

Thank you!

**Comment by rneher** [comment]
thanks. we'll look into the Singapore data, but we would need a parser for these data and basic population statistics as described here: https://github.com/neherlab/covid19_scenarios/blob/master/data/README.md#contributing-and-curating-data

could you contribute such data and parsers?

**Comment by hxchua** [comment, marked as answer]
Yes, I can create the parser after your confirmation on what should be
included in the hospitalised count. Will it be alright if I can only
provide case count data but not population data? I noticed that
the populationData.tsv dataset already contains Singapore. Thanks!

**Reply by rneher** [reply]
yes, you are right. we have Singapore data through the ecdc data set. Our model doesn't fit hospitalization data right now, but they are displayed on the graph. The models count for hospitalized cases is equivalent to those severely ill. So home isolation would not be part of this.

**Reply by hxchua** [reply]
Got it, thanks. Sorry another thing to confirm, is ICU part (i.e. subset) or independent of hospitalized?

**Reply by rneher** [reply]
We think of them as separate and more in terms of regular and ICU ward.

**Comment by hxchua** [comment]
When I tried to run the parser, I ran into this error: ModuleNotFoundError: No module named 'generated'.

In util.py, there's a line on `import generated.types as schema`. Appreciate advice on where does the module come from?

**Reply by ivan-aksamentov** [reply]
@hxchua Looks like you don't have the required Python classes that we generate from schemas (`make types`).
Typically, in order to run the parser step (download data and save to TSV files, `make tsv-data`) you don't need these classes. However, you need them in order to run the converter step (which generates the final JSON file that will be consumed by the application, `make cases`). Is it converter what you are trying to run?

Here is a quick guide:

We share type definitions between the data generators (Python) and the application itself (Typescript). In order to ensure strong typing across languages, we have JSON-schemas in [`schemas/`](https://github.com/neherlab/covid19_scenarios/tree/5fdbf45ed68dad2112dd794e8de46e4eee3f914b/schemas) directory. There is a script [`tools/generateTypes.ts`](https://github.com/neherlab/covid19_scenarios/blob/5fdbf45ed6/tools/generateTypes.ts) which takes these schemas and converts them to Python classes, Typescript typings and Javascript ajv validators. You need Node.js in order to run it. If you have Node.js you can run [`yarn schemas:totypes`](https://github.com/neherlab/covid19_scenarios/blob/5fdbf45ed68dad2112dd794e8de46e4eee3f914b/package.json#L11)
from the project root, or run `make types` in data directory, which will do exactly the same:
https://github.com/neherlab/covid19_scenarios/blob/5fdbf45ed68dad2112dd794e8de46e4eee3f914b/data/Makefile#L11-L12

After that the `data/generated/types.py` file will appear, and the things should start working.
Take a look at other commands in this Makefile if you want to generate the data for the application.
